pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java

```diff
@@ -272,7 +272,13 @@ public static int getPartitionIndex(String topic) {
         int partitionIndex = -1;
         if (topic.contains(PARTITIONED_TOPIC_SUFFIX)) {
             try {
-                partitionIndex = Integer.parseInt(topic.substring(topic.lastIndexOf('-') + 1));
+                String idx = StringUtils.substringAfterLast(topic, PARTITIONED_TOPIC_SUFFIX);
+                partitionIndex = Integer.parseInt(idx);
```

Comment:
Should we check `partitionIndex >= 0` and return -1 as the error index? You can add a test case for `mytopic-partition--2`.